src/js/background/assignManager.js

```diff
@@ -282,11 +316,19 @@ const assignManager = {
   },
 
   async _onClickedBookmark(info) {
-    let bookmarks = await browser.bookmarks.get(info.bookmarkId);
-    if (bookmarks[0].type === "folder") { 
-      bookmarks = await browser.bookmarks.getChildren(bookmarks[0].id);
+
+    async function _getBookmarksFromInfo(info) {
+      const bookmarkTreeNode = await browser.bookmarks.get(info.bookmarkId);
+      const firstBookmark = bookmarkTreeNode[0];
+      if (firstBookmark.type === "folder") {
+        return await browser.bookmarks.getChildren(firstBookmark.id);
+      } else {
```

Comment:
You can drop the `else` logic branch here, because the preceding line will return out of the function early. So you can put the `return bookmarkTreeNode` after the `if (firstBookmark.type === "folder") {}` block.

Reply:
I find that using `const` variables and keeping my logic branches to a minimum helps keep the code straight-forward.